src/Controls/src/Core/Handlers/Items/iOS/ItemsViewController.cs

```diff
@@ -200,19 +200,45 @@ public override void ViewWillLayoutSubviews()
 		{
 			ConstrainItemsToBounds();
 
-			if (CollectionView is Items.MauiCollectionView { NeedsCellLayout: true } collectionView)
+			var mauiCollectionView = CollectionView as MauiCollectionView;
+			var needsCellLayout = mauiCollectionView?.NeedsCellLayout is true;
+			if (needsCellLayout)
 			{
 				InvalidateLayoutIfItemsMeasureChanged();
-				collectionView.NeedsCellLayout = false;
+				mauiCollectionView.NeedsCellLayout = false;
 			}
 
 			base.ViewWillLayoutSubviews();
+
+			if (needsCellLayout || !_laidOut)
+			{
+				var isRefreshing = CollectionView.Subviews
+					.OfType<UIRefreshControl>()
+					.FirstOrDefault()?.Refreshing ?? false;
```

Comment:
Performance implications of LINQ are a bit unknown right now

Can we remove the use of LINQ here?